src/error.rs

```diff
@@ -0,0 +1,38 @@
+#[derive(thiserror::Error, Debug)]
+pub enum PlRustError {
```

Comment:
This should be split up later once the crate has more structure, but it's a start.